src/components/UserRegister.tsx

```diff
@@ -23,227 +26,180 @@ const Flags = [
-      marginLeft: '10px',
-      width: '96px',
-      height: '30px',
-    },
-
-    formnames: {
-      width: '170px',
-      height: '58px',
-      marginLeft: '10px',
-      marginBottom: '20px',
-    },
-    formfields: {
-      width: '366px',
-      height: '58px',
-    },
-    phonenumber: {
-      width: '366px',
-      height: '58px',
-
-      '& .MuiOutlinedInput-input:-webkit-autofill': {
-        paddingLeft: '44px',
-      },
-      '& .MuiOutlinedInput-input': {
-        paddingLeft: '41px',
-      },
-    },
-
-    container: {
-      position: 'relative',
-      padding: '0',
-      margin: '0',
-
-      '& .MuiFormLabel-root.Mui-focused': {
-        paddingLeft: '5px',
-      },
-      '& .MuiInputLabel-outlined': {
-        paddingLeft: '30px',
-      },
-    },
-    flagimg: {
-      position: 'absolute',
-      bottom: '20px',
-      left: '12px',
-      width: '24px',
-      height: '16px',
-    },
-  })
-);
-
 const UserForm = (props: any) => {
   let history = useHistory();
-  const classes = useStyles();
-  const formClasses = formStyles();
   //This state is to send true or false to component FlagSelect to trigger dialogue box
   const [isFlagDialog, setisFlagDialog] = useState(false);
   const [selectedFlag, setSelectedFlag] = React.useState(Flags[0].img);
-  const [phonePattern, setPhonePattern] = useState('US');
-  const [loader, setLoader] = useState(false);
+  const [loaded, setLoaded] = useState(true);
+  const resolver = yupResolver(signUpSchema);
 
   useEffect(() => {
     return () => setisFlagDialog(false);
   }, []);
 
-  const {
-    register,
-    handleSubmit,
-    formState: { errors, isValid },
-  } = useForm<IFormInput>({ mode: 'onChange' });
+  const { handleSubmit, formState, control, setValue } = useForm<any>({ mode: 'onBlur', resolver });
+  const { isValid } = formState;
 
   const onSubmit = (data: IFormInput) => {
-    setLoader(true);
-    setTimeout(() => {
-      setLoader(false);
-    }, 5000);
-
+    setLoaded(!loaded);
     data = {
       ...data,
       firstName: capitalizeFirstLetter(data.firstName),
-      lastName: capitalizeFirstLetter(data.lastName),
+      // lastName: capitalizeFirstLetter(data.lastName),
     };
     props.submitUser(data);
-    history.push(path.HOME_PAGE);
+
+    setTimeout(() => {
+      setLoaded(true);
+      history.push(path.HOME_PAGE);
+    }, 3000);
   }; // your form submit function which will invoke after successful validation
 
-  //Call back function for FlagSelect component
+  // Call back function for FlagSelect component
   const flagStateChange = (index: number) => {
     setisFlagDialog(false);
     setSelectedFlag(Flags[index].img);
-    Flags[index].code === '+1-United States' ? setPhonePattern('US') : setPhonePattern('AUS');
+    // Created a custum field in register validate according to country in yup schema
+    Flags[index].code === '+1-United States' ? setValue('country', countries.AMERICA) : setValue('country', countries.AUSTRALIA);
```

Comment:
ternary operator is used where you have to return something
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator

Reply:
Got it done